src/components/interaction-designer/clipboard/ClipboardRoot.vue

```diff
@@ -0,0 +1,154 @@
+      </div>
+      <div v-if="unsupportedBlockName" class="mt-2">
+        <UnsupportedBlock :block-name="unsupportedBlockName"  />
+      </div>
+    </main>
+    <footer v-if="isComplete" class="mt-2">
+      <div class="card">
+        <div class="card-body">
+          <h5 class="card-title">{{'flow-builder.completed' | trans}}!</h5>
+        </div>
+      </div>
+    </footer>
+  </div>
+</template>
+
+<script lang="ts">
+import lang from '@/lib/filters/lang'
+import { mapActions, mapGetters } from 'vuex'
+import {
+  IContext,
+  createContextDataObjectFor,
+  FlowRunner,
+  IRichCursorInputRequired,
+  SupportedMode,
+  IContact,
+} from '@floip/flow-runner'
+import MessageBlock from './block/MessageBlock.vue'
+import NumericQuestionBlock from './block/NumericQuestionBlock.vue'
+import OpenQuestionBlock from './block/OpenQuestionBlock.vue'
+import SelectOneResponseBlock from './block/SelectOneResponseBlock.vue'
+import SelectManyResponseBlock from './block/SelectManyResponseBlock.vue'
+import UnsupportedBlock from './shared/UnsupportedBlock.vue'
+
+export default {
+  components: {
+    MessageBlock,
+    NumericQuestionBlock,
+    OpenQuestionBlock,
+    SelectOneResponseBlock,
+    SelectManyResponseBlock,
+    UnsupportedBlock,
+  },
+  mixins: [lang],
+  data() {
+    return {
+      runner: FlowRunner,
+      blockPrompts: [],
+      context: {},
+      isComplete: false,
+      unsupportedBlockName: '',
+    }
+  },
+  created() {
+    this.initializeFlowRunner()
+  },
+  methods: {
+    ...mapGetters('flow', ['getFlowState']),
+    ...mapActions('clipboard', ['setSimulatorActive']),
+
+    async initializeFlowRunner() {
+      const flowState = this.getFlowState()
+      const contact = { id: '1' } as IContact
+      const groups = []
+      const userId = 'user-1234'
+      const orgId = 'org-1234'
+      const { flows } = flowState
+      const languageId = '22'
+      const mode = SupportedMode.OFFLINE
+      const { resources } = flowState
+      const context: IContext = createContextDataObjectFor(
+        contact,
+        groups,
+        userId,
+        orgId,
+        flows,
+        languageId,
+        mode,
+        resources,
+      )
+
+      this.context = context
+      this.runner = new FlowRunner(context)
+      await this.goNext()
+    },
+
+    async goNext() {
+      this.isComplete = false
+      try {
+        const cursor: IRichCursorInputRequired = await this.runner.run()
+        if (!cursor) {
+          this.isComplete = true
+          return
+        }
+        const { prompt }: IRichCursorInputRequired = cursor
+        this.blockPrompts.push(prompt)
+      } catch (e) {
+        if (e.message.includes('Unable to find factory for block type')) {
+          [, this.unsupportedBlockName] = e.message.split(': ')
+        }
+        console.warn(e.message)
+      }
+    },
+
+    getBlockComponent(key) {
+      const blockMap = new Map([
+        ['MobilePrimitives\\Message', 'MessageBlock'],
```

Comment:
I think I missed a piece when I was drafting AC, my apologies. 

---

Prompts have a `kind` property, which is an ambiguated form of the block types to specialize on the type of data being collected from the contact, rather than how the runner should _handle_ that type of data. 

An example: 
_We could have two specific block types that both need to collect Numeric input from the contact, each runner could do something wildly different with the data once it's been collected, but the phase of Prompt-ing the user for input and validating that input will be the same and can be reused between them._

What I'd like to propose is that we go through and refactor this a bit to instead pivot over Prompt `kind`. This allows us to ignore block specialization and keep us focused on the reduced set that revolves around the type of data we're collecting from the contact. 

Changes I can think of:
- rename from clipboard block types to clipboard prompts
- update this from having an explicit mapping of block type, to rather look up the kind in configured `Prompt.VALUES` on current runnner (via `flow-runner/src/domain/prompt/Prompt.ts`) — We can hash this one out together, it is not supper straightforward, but a list of configured prompt kinds can be retrieved/used like:
```
import {map, some} from lodash

const knownPromptKinds = map(Prompt.VALUES, 'promptKey')
// < ['Message', 'Numeric', 'SelectOne', 'SelectMany', 'Open']

const promptKindToFind = 'Numeric'
const isValidPromptKind = some(Prompt.VALUES, ({promptKey: kind}) => kind === promptKindToFind)
// < true
```

Take a peek at some documentation I'd written up on prompts on the github repo; I've outlined the prompt kinds specifically here: https://github.com/FLOIP/flow-runner#prompt-types — but it looks like more have been added to the repo since then.

---

CC: @safydy — this is partially a response to: https://github.com/FLOIP/flow-builder/pull/39/files?diff=unified&w=1#r596348051

Reply:
Fixed!